internal/store/attempt_repo.go

```
@@ -0,0 +1,607 @@
+func expiredFinishAuthorization(_ FinishAttemptParams) (string, []any) {
+	return "julianday(lease_expires_at) <= julianday('now')", nil
+}
+
+func NewAttemptRepo(db *DB) *AttemptRepo {
+	return &AttemptRepo{db: db}
+}
+
+// GetResumeCandidate snapshots task status and its latest terminal run so a
+// caller cannot accidentally adopt a concurrent invocation's running attempt.
+func (r *AttemptRepo) GetResumeCandidate(taskID int64) (*ResumeCandidate, error) {
+	row := r.db.sql.QueryRow(
```

golangci-lint (2.12.2)

[error] (*database/sql.DB).QueryRow must not be called. use (*database/sql.DB).QueryRowContext

(noctx)